site/app/controllers/forum/ForumController.php

```diff
@@ -879,7 +897,21 @@ private function editPost() {
 
             $markdown = !empty($_POST['markdown_status']);
 
-            return $this->core->getQueries()->editPost($original_creator, $current_user, $post_id, $new_post_content, $anon, $markdown);
+            $current_attachments = explode("\n", $original_post['attachment_name']);
+            if (isset($_POST['deleted_attachments'])) {
+                foreach (json_decode($_POST['deleted_attachments']) as $img) {
```

Comment:
Suggestion: you could get rid of one level of nesting here by doing something like the following.  (whether or not this is a good idea or not is up to you to decide)
```suggestion
            foreach (json_decode($_POST['deleted_attachments'] ?? '[]') as $img) {
```